text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+# Context Propagation: A Layered Approach 
+
+* [Motivation](#Motivation)
+* [OpenTelemetry layered architecture](#OpenTelemetry-layered-architecture)
+  * [Cross-Cutting Concerns](#Cross-Cutting-Concerns)
+    * [Observability API](#Observability-API)
+    * [Correlations API](#Correlations-API)
+  * [Context Propagation](#Context-Propagation)
+    * [Context API](#Context-API)
+    * [Propagation API](#Propagation-API)
+* [Prototypes](#Prototypes)
+* [Examples](#Examples)
+  * [Global initialization](#Global-initialization)
+  * [Extracting and injecting from HTTP headers](#Extracting-and-injecting-from-HTTP-headers)
+  * [Simplify the API with automated context propagation](#Simplify-the-API-with-automated-context-propagation)
+  * [Implementing a propagator](#Implementing-a-propagator)
+  * [Implementing a concern](#Implementing-a-concern)
+  * [The scope of current context](#The-scope-of-current-context)
+  * [Referencing multiple contexts](#Referencing-multiple-contexts)
+  * [Falling back to explicit contexts](#Falling-back-to-explicit-contexts)
+* [Internal details](#Internal-details)
+* [FAQ](#faq)
+
+Refactor OpenTelemetry into a set of separate cross-cutting concerns which 
+operate on a shared context propagation mechanism.
+
+# Motivation
+
+This RFC addresses the following topics:
+
+**Separation of concerns**  
+* Remove the Tracer dependency from context propagation mechanisms.
+* Handle user data (Correlations) and observability data (SpanContext, etc) 
+  separately.  
+
+**Extensibility**
+* Allow developers to create new applications for context propagation. For 
+  example: A/B testing, encrypted or authenticated data, and new, experimental 
+  forms of observability.
+
+
+# OpenTelemetry layered architecture
+
+The design of OpenTelemetry is based on the principles of [aspect-oriented 
+programming](https://en.wikipedia.org/wiki/Aspect-oriented_programming), 
+adopted to the needs of distributed systems. 
+
+Some concerns "cut across" multiple abstractions in a program. Logging 
+exemplifies aspect orientation because a logging strategy necessarily affects 
+every logged part of the system. Logging thereby "cross-cuts" across all logged 
+classes and methods. Distributed tracing takes this strategy to the next level, 
+and cross-cuts across all classes and methods in all services in the entire 
+transaction. This requires a distributed form of the same aspect-oriented 
+programming principles in order to be implemented cleanly.
+
+OpenTelemetry approaches this by separating it's design into two layers. The top 
```

Comment:
is it desired state or what's implemented now? I'm not following

Reply:
What is happening now. I wrote this with the intention of copy-pasting into the spec. I can change the tense if you prefer.